elki/src/main/java/de/lmu/ifi/dbs/elki/algorithm/itemsetmining/Itemset.java

```diff
@@ -76,6 +97,13 @@ public int getSupport() {
    * @return Itemset contents.
    */
   abstract public long[] getItems();
+  
+  /**
+   * Get a SparseItemset like representation of the Itemset.
+   * 
+   * @return Itemset contents.
+   */
+  abstract public int[] toSparseRep();
```

Comment:
Rather than always forcing everything into `int[]`, it would be better to work with the `Itemset` objects directly, similar to the `SparseNumberVector` API, which is very memory efficient.